MobiusCore/Source/MobiusController.swift

```diff
@@ -33,36 +33,67 @@ public final class MobiusController<Model, Event, Effect> {
     private let state: State
 
     /// A Boolean indicating whether the MobiusLoop is running or not.
-    ///
-    /// May not be called directly from the update function or an effect handler running on the controller’s loop queue.
     public var running: Bool {
         return state.running
     }
 
+    /// See `Mobius.Builder.makeController` for documentation
     init(
         builder: Mobius.Builder<Model, Event, Effect>,
         initialModel: Model,
-        loopQueue: DispatchQueue,
+        loopQueue loopTargetQueue: DispatchQueue,
         viewQueue: DispatchQueue
     ) {
-        let actualLoopQueue = DispatchQueue(label: "MobiusController \(Model.self)", target: loopQueue)
+        // The internal loopQueue is a serial queue targeting the provided queue, so that targeting a concurrent queue
+        // doesn’t result in concurrent work on the underlying MobiusLoop.
+        let loopQueue = DispatchQueue(label: "MobiusController \(Model.self)", target: loopTargetQueue)
         self.viewQueue = viewQueue
 
-        let state = State(model: initialModel, queue: actualLoopQueue)
+        let state = State(model: initialModel, queue: loopQueue)
         self.state = state
-
+        /*
+         Note: `flipEventsToLoopQueue` should not and in fact cannot capture `self`, because it’s used to initialize
+         `self.loopFactory`. It can however capture `state`.
+
+         In fact, here’s the ownership graph after initing:
+                     ┏━━━━━━━━━━━━┓
+                ┌────┨ controller ┠────────┐
```

Comment:
Nice graph! The ASCII graphics are a little messed up in the GHE browser; I assume they look fine in XCode?

Reply:
Looks fine in my Github :-/

The vertical spacing is a bit nicer in Xcode, unless of course you use a proportional font for comments. My other option was to embed some Graphviz code…